docs/architecture/adr-069-flexible-node-intitalization.md

```diff
@@ -0,0 +1,164 @@
+    several interfaces and constructors, or simply a function in the
+    node package. 
+
+- Refactor and simplify the process start/initialization logic,
+  separately from--but as a precursor to--permitting user replacement
+  of components.
+
+## Non-Goals
+
+- Fully abstract dependency-based process initialization. 
+
+- Fully abstract replacement of core tendermint components without vendoring.
+
+- Parallelized initialization of components.
+
+## Questions
+
+- To what extent does this new initialization framework need to accommodate
+  the legacy p2p stack? Would it be possible to delay a great deal of this
+  work to the 0.36 cycle to avoid this complexity? 
+  
+  - Answer: _depends on timing_, and the requirement to ship pluggable
+    reactors in 0.35. 
+  
+- There's clear and common interest in injecting some kind of mempool, and
+  intuitively this makes sense as the highest priority.
+  
+  - Does the existing `mempool.Mempool` interface make sense to expose and
+    commit to, or should we make a smaller interface? 
+    
+	- Answer: _yes_, smaller mempool interface is better. 
+
+  - Is it reasonable for users to bring their own mempool without bringing
+    their own reactor?
+	
+	- Answer: _probably not_, but this might be a function of how the
+      current implementations work and can be refactored. 
+
+- Are there reactors, other than the mempool, that make sense to support
+  ad-hoc replacing (e.g. statesync, blockchain (fastsync),) and if so should
+  we provide any first-class support for this?
+
+  - Answer: _defer_. Certainly this shouldn't be attempted until after
+    there is a refactor of the consensus reactor. Is more likely than
+    block/fast sync. 
+
+- There's a dependency (and nearly a cycle), between consensus,
+  blockchain (fastsync), which makes it very hard to allow meaningful
+  injection of any of these components without allowing injection of
+  all of them (statesync provider as a possible exception.) Is this
+  acceptable?
+  
+  - Answer: _yes_. 
+
+## Future Work
+
+- Improve or simplify the `service.Service` interface. There are some
+  pretty clear limitations with this interface as written (there's no
+  way to timeout slow startup or shut down, the cycle between the
+  `service.BaseService` and `service.Service` implementations is
+  troubling, the default panic in `OnReset` seems troubling.)
+
+- As part of the refactor of `service.Service` have all services/nodes
+  respect the lifetime of a `context.Context` object, and avoid the
+  current practice of creating `context.Context` objects in p2p and
+  reactor code. This would be required for in-process multi-tenancy.
+  
+- Consider additional interfaces that node objects can provide in
+  terms of access to components (e.g. the mempool).
+
+- Support explicit dependencies between components and allow for
+  parallel startup, so that different reactors can startup at the same
+  time, where possible.
+
+## Proposal / Conclusion 
```

Comment:
I'm not entirely sure what this ADR describes as far as the implementation. Usually, ADR's have short excerpts of a proposed design.